src/main/java/org/tugraz/sysds/runtime/controlprogram/context/SparkExecutionContext.java

```diff
@@ -317,19 +317,19 @@ public static boolean isLocalMaster() {
 	 * variables.
 	 *
 	 * @param varname variable name
-	 * @return JavaPairRDD of TensorIndexes-TensorBlocks
+	 * @return JavaPairRDD of TensorIndexes-HomogTensors
 	 */
 	@SuppressWarnings("unchecked")
-	public JavaPairRDD<TensorIndexes, TensorBlock> getBinaryTensorBlockRDDHandleForVariable(String varname ) {
+	public JavaPairRDD<TensorIndexes, HomogTensor> getBinaryTensorBlockRDDHandleForVariable(String varname ) {
 		TensorObject to = getTensorObject(varname);
-		return (JavaPairRDD<TensorIndexes, TensorBlock>)
+		return (JavaPairRDD<TensorIndexes, HomogTensor>)
 				getRDDHandleForTensorObject(to, InputInfo.BinaryTensorBlockInputInfo, -1, true);
 	}
 
 	@SuppressWarnings("unchecked")
-	public JavaPairRDD<TensorIndexes, TensorBlock> getBinaryTensorBlockRDDHandleForVariable(String varname, int numParts, boolean inclEmpty ) {
+	public JavaPairRDD<TensorIndexes, HomogTensor> getBinaryTensorBlockRDDHandleForVariable(String varname, int numParts, boolean inclEmpty ) {
```

Comment:
Again, this should return TensorBlock as otherwise we would have to introduce the same thing for heterogeneous tensors.